core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -411,7 +411,15 @@ class ControllerBrokerRequestBatch(controller: KafkaController) extends  Logging
           .format(broker, stopReplicaWithDelete.mkString(",")))
         debug("The stop replica request (delete = false) sent to broker %d is %s"
           .format(broker, stopReplicaWithoutDelete.mkString(",")))
-        replicaInfoList.foreach { r =>
+
+        // Send one StopReplicaRequest for all partitions that requires neither delete nor callback
+        val stopReplicaRequest = new StopReplicaRequest(controllerId, controllerEpoch, false,
```

Comment:
`StopReplicaRequest` is a Java class and it doesn't allow us to use named arguments, right?